src/plugins/data/common/index_patterns/index_patterns/index_pattern.ts

```diff
@@ -519,8 +524,13 @@ export class IndexPattern implements IIndexPattern {
             // and ensure we ignore the key if the server response
             // is the same as the original response (since that is expected
             // if we made a change in that key)
-            const serverChangedKeys = Object.keys(updatedBody).filter((key) => {
-              return updatedBody[key] !== body[key] && this.originalBody[key] !== updatedBody[key];
+
+            const serverChangedKeys: string[] = [];
+            Object.entries(updatedBody).forEach(([key, value]) => {
+              // @ts-ignore
```

Comment:
I think these two ts-ignore statements are an acceptable compromise for now. It would be nice to do a deeper refactoring on this code first.